pkg/sql/insert_fast_path.go

```diff
@@ -343,3 +346,43 @@ func (n *insertFastPathNode) Close(ctx context.Context) {
 func (n *insertFastPathNode) enableAutoCommit() {
 	n.run.ti.enableAutoCommit()
 }
+
+// interceptAlterColumnTypeParseError wraps a type parsing error with a warning
+// about the column undergoing an ALTER COLUMN TYPE schema change.
+// If colNum is not -1, only the colNum'th column in insertCols will be checked
+// for AlterColumnTypeInProgress, otherwise every column in insertCols will
+// be checked.
+func interceptAlterColumnTypeParseError(
```

Comment:
I feel like all the duplicated nesting this function does could be deduplicated

Reply:
Made some changes to this function to deduplicate code, let me know what you think.